client.go

```diff
@@ -190,6 +190,26 @@ func (c *Client) trigger(channels []string, event string, data interface{}, sock
 	return unmarshalledBufferedEvents(response)
 }
 
+type BatchRequest struct {
+	Batch []Event `json:"batch"`
+}
+
+func (c *Client) TriggerBatch(batch []Event) (*BufferedEvents, error) {
```

Comment:
In the case of `event` vs `Name` I think it's more `event` vs `event.Name` so I would rename the first one to `eventName`.

I'm not sure if we can compact all the trigger methods. `trigger(Event[]{{"event", "channel", "data"}})` doesn't seem much of an improvement, and it doesn't support the multi-channel scenario.